### ISSUE TITLE
When using Ollama, a long loop of Initializing Ollama Model happens before any output

### ISSUE THREAD

**Issue by FGhrawi** [elizaOS/eliza#443]
**Describe the bug**

When using Ollama, the model being used is loaded over and over in a loop (sometimes 10-15+ times) in the terminal before any output or agent decision-making happens.
Fwiw, one model is configured at a time in .env (so no swapping is being done)

**To Reproduce**

Use any model with the Ollama provider

**Expected behavior**

Model loads once. (Since ollama is on keepalive and only 1 model is being used)

**Screenshots**
![Screenshot_2024-11-19_at_10 46 50_PM](https://github.com/user-attachments/assets/9aee0e5e-a587-482e-9afc-b860b1a23614)

**Additional context**
I am on a fresh clone of the repo with a basic character configuration.
Fwiw, I have 24gb vram, and this happens even on smaller models.

**Comment by yodamaster726** [comment]
I did a fresh clone on the latest code and the code keeps wanting to download and run the local_llama.
I have the OLLAMA_MODEL set in my .env.  Here is what have:

#Set to Use for New OLLAMA provider
OLLAMA_SERVER_URL=   #Leave blank for default localhost:11434
OLLAMA_MODEL=hermes3
OLLAMA_EMBEDDING_MODEL=     #default mxbai-embed-large
#To use custom model types for different tasks set these
SMALL_OLLAMA_MODEL=         #default llama3.2
MEDIUM_OLLAMA_MODEL=        #default hermes3
LARGE_OLLAMA_MODEL=         #default hermes3:70b

This is what I have in my defaultCharacter.

    modelProvider: ModelProviderName.OLLAMA,
    settings: {
        secrets: {},
        voice: {
            model: "en_US-hfc_female-medium",
        },
        embeddingModel: "mxbai-embed-large"
    },

Code is also tries to load the wrong OLLAMA embedding model too by default - hence having to set it here.

I'm debugging the problem now and working on a fix for it.

**Comment by lakshya404stc** [comment]
Is this issue still valid??

**Comment by yodamaster726** [comment]
yes, ollama and llama local got merged together and the ollama logic is not working right.  getting close to a fix.
example, its downloading the local llama instead of using ollama if it's configured

**Comment by drew-royster** [comment]
@yodamaster726 what is your character file like? I had ollama as the model in my character and it seemed to use ollama just fine.

**Comment by yodamaster726** [comment]
I tried the latest tag v0.1.3 and then tried the latest code from yesterday.  My character file was the default one.
![image](https://github.com/user-attachments/assets/140300b3-dc4e-4cb1-89f7-810f4589c627)

updates to fix this problem:  https://github.com/ai16z/eliza/pull/521

**Comment by dr-fusion** [comment]
just tested with the latest code and used `ModelProviderName.LLAMALOCAL,`

Seems like its still going in a loop.

**Comment by MERNinja** [comment]
ModelProviderName.LLAMALOCAL is still in loop with alpha.1 tag release. Anyone found a solution?

**Comment by AIFlowML** [comment]
> ModelProviderName.LLAMALOCAL is still in loop with alpha.1 tag release. Anyone found a solution?

Did you try last version we just release ?
If is not working at all i wil look in the code.
Please let me know.

**Comment by Luucky** [comment]
I tried the following fix locally which prevented the looping on LLAMA_LOCAL model provider
https://github.com/elizaOS/eliza/pull/1755/files

**Comment by AIFlowML** [comment]
The solution presented by @Luucky is the most viable.
Try update to the new version please.
